File: test-runtime/src/main/java/dev/vibris/testruntime/FakeRuntimeProbe.java
package dev.vibris.testruntime;

import dev.vibris.core.CoreProbe;
import dev.vibris.protocol.v1.SceneContext;

import java.util.List;

public final class FakeRuntimeProbe {
    private final FakeRuntimeAdapter runtime;
    private final CoreProbe core;

    FakeRuntimeProbe(FakeRuntimeAdapter runtime, CoreProbe core) {
        this.runtime = runtime;
        this.core = core;
    }

    public void pauseExecution() {
        runtime.pauseExecution();
    }

    public void resumeExecution() {
        runtime.resumeExecution();
    }

    public List<String> executionOrder() {
        return core.executionOrder();
    }

    public List<String> executionEvents() {
        return core.executionEvents();
    }

    public List<SceneContext> contextSnapshots() {
        return core.contextSnapshots();
    }

    public List<String> sourceStates(String uuid) {
        return core.sourceStates(uuid);
    }

    public int maxConcurrentJobs() {
        return core.maxConcurrentJobs();
    }

    public int executionCount(String requestId) {
        return core.executionCount(requestId);
    }

    public int requestRegistrySize() {
        return core.requestRegistrySize();
    }

    public int requestRegistryCapacity() {
        return core.requestRegistryCapacity();
    }

    public int sourceRegistrySize() {
        return core.sourceRegistrySize();
    }

    public int sourceRegistryCapacity() {
        return core.sourceRegistryCapacity();
    }

    public int queueSize() {
        return core.queueSize();
    }

    public int queueCapacity() {
        return core.queueCapacity();
    }

    int maxConcurrentRuntimeOperations() {
        return runtime.maxConcurrentOperations();
    }

    CoreProbe.Snapshot snapshot() {
        return core.snapshot();
    }
}

File: test-runtime/src/main/kotlin/dev/vibris/testruntime/FakeRuntimeProbe.kt
package dev.vibris.testruntime

import dev.vibris.core.CoreProbe
import dev.vibris.protocol.v2.SceneContext

class FakeRuntimeProbe internal constructor(
    private val runtime: FakeRuntimeAdapter,
    private val core: CoreProbe,
) {
    fun pauseExecution() {
        runtime.pauseExecution()
    }

    fun resumeExecution() {
        runtime.resumeExecution()
    }

    fun executionOrder(): List<String> = core.executionOrder()

    fun executionEvents(): List<String> = core.executionEvents()

    fun contextSnapshots(): List<SceneContext> = core.contextSnapshots()

    fun sourceStates(uuid: String): List<String> = core.sourceStates(uuid)

    fun maxConcurrentJobs(): Int = core.maxConcurrentJobs()

    fun executionCount(requestId: String): Int = core.executionCount(requestId)

    fun requestRegistrySize(): Int = core.requestRegistrySize()

    fun requestRegistryCapacity(): Int = core.requestRegistryCapacity()

    fun sourceRegistrySize(): Int = core.sourceRegistrySize()

    fun sourceRegistryCapacity(): Int = core.sourceRegistryCapacity()

    fun queueSize(): Int = core.queueSize()

    fun queueCapacity(): Int = core.queueCapacity()

    internal fun maxConcurrentRuntimeOperations(): Int = runtime.maxConcurrentOperations()

    internal fun snapshot(): CoreProbe.Snapshot = core.snapshot()
}